pyvista/core/grid.py

```diff
@@ -250,70 +253,109 @@ def cast_to_structured_grid(self):
 class UniformGrid(_vtk.vtkImageData, Grid, UniformGridFilters):
     """Models datasets with uniform spacing in the three coordinate directions.
 
-    Can be initialized in several ways:
+    Can be initialized in one of several ways:
 
     - Create empty grid
     - Initialize from a vtk.vtkImageData object
-    - Initialize directly from the point arrays
+    - Initialize based on dimensions, cell spacing, and origin.
 
-    See ``_from_specs`` in the documentation for more details on initializing
-    from point arrays
+    .. versionchanged:: 0.33.0
+        First argument must now be either a path or
+        ``vtk.vtkImageData``. Use keyword arguments to specify the
+        dimensions, spacing, and origin of the uniform grid.
+
+    Parameters
+    ----------
+    uinput : str, vtk.vtkImageData, pyvista.UniformGrid, optional
+        Filename or dataset to initialize the uniform grid from.  If
+        set, remainder of arguments are ignored.
+
+    dims : iterable, optional
+        Dimensions of the uniform grid.
+
+    spacing : iterable, optional
+        Spacing of the uniform in each dimension.  Defaults to
+        ``(1.0, 1.0, 1.0)``.
+
+    origin : iterable, optional
+        Origin of the uniform grid.  Defaults to ``(0.0, 0.0, 0.0)``.
 
     Examples
     --------
-    >>> import pyvista
-    >>> import vtk
-    >>> import numpy as np
+    Create an empty UniformGrid.
 
-    >>> # Create empty grid
+    >>> import pyvista
     >>> grid = pyvista.UniformGrid()
 
-    >>> # Initialize from a vtk.vtkImageData object
+    Initialize from a ``vtk.vtkImageData`` object.
+
+    >>> import vtk
     >>> vtkgrid = vtk.vtkImageData()
     >>> grid = pyvista.UniformGrid(vtkgrid)
 
-    >>> # Using just the grid dimensions
-    >>> dims = (10, 10, 10)
-    >>> grid = pyvista.UniformGrid(dims)
+    Initialize using using just the grid dimensions and default
+    spacing and origin.
+
+    >>> grid = pyvista.UniformGrid(dims=(10, 10, 10))
 
-    >>> # Using dimensions and spacing
-    >>> spacing = (2, 1, 5)
-    >>> grid = pyvista.UniformGrid(dims, spacing)
+    Initialize using dimensions and spacing.
 
-    >>> # Using dimensions, spacing, and an origin
-    >>> origin = (10, 35, 50)
-    >>> grid = pyvista.UniformGrid(dims, spacing, origin)
+    >>> grid = pyvista.UniformGrid(
+    ...     dims=(10, 10, 10),
+    ...     spacing=(2, 1, 5),
+    ... )
+
+    Initialize using dimensions, spacing, and an origin.
+
+    >>> grid = pyvista.UniformGrid(
+    ...     dims=(10, 10, 10),
+    ...     spacing=(2, 1, 5),
+    ...     origin=(10, 35, 50),
+    ... )
 
     """
 
     _WRITERS = {'.vtk': _vtk.vtkDataSetWriter, '.vti': _vtk.vtkXMLImageDataWriter}
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
```

Comment:
Based on the description:

> Use keyword arguments to specify the
        dimensions, spacing, and origin of the uniform grid.

It would be my preference to require that they are keyword args, but I don't feel strongly about it.  If so, it should be done gracefully with a deprecation


```py
def __init__(
            self,
            uinput=None,
            *args,
            dims=None,
            spacing=(1.0, 1.0, 1.0),
            origin=(0.0, 0.0, 0.0)
    ):

    # handle case with dims, spacing, and/or origin
    # with deprecation.
```

Reply:
Agreed and done in cec6dce1b3d7a3a45eb67020a49bb7506e1eb95d and b96458657a34cf9572a3bf99dc214de6aa93f185. Added deprecation check in our unit testing.